source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.h

```diff
@@ -103,11 +103,21 @@ class JsonTranscoderConfig : public Logger::Loggable<Logger::Id::config>,
    */
   bool convertGrpcStatus() const;
 
+  /**
+   * @param buffer_size the size of the buffer, returned if config did not specify a size.
+   * @return the maximum size of the message body.
+   */
+  uint32_t maxBodySize(uint32_t buffer_size) const {
+    return max_body_size_ == 0 ? buffer_size : max_body_size_;
```

Comment:
Should this be max(buffer_size, max_body_size_)?

Reply:
No, max_(request|response)_body_size_ can validly argue for accepting a smaller message size than the size of the stream buffer.